src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -310,6 +310,31 @@ public static TimeSpan DefaultTimeout
                 }
                 return s_defaultTimeout;
             }
+            set
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceBase, "TransactionManager.set_DefaultTimeout");
+                }
+
+                s_defaultTimeout = ValidateTimeout(value);
+
+                if (s_defaultTimeout != DefaultSettingsSection.Timeout)
+                {
```

Comment:
The method `ValidateTimeout` can return a value lower than the passed value. If the value passed is larger than MaximumTimeout, then it returns MaximumTimeout. The event here is intended to emit an etw event if the resulting timeout is different than the requested one. The comparison should be `s_defaultTimeout != value` to trigger the event when `ValidateTimeout` returned something different than what was passed in.

Reply:
Thanks for pointing it out! That makes sense. I have updated code.